stake-pool/js/src/schema.ts

```diff
@@ -0,0 +1,167 @@
+import borsh from 'borsh';
+import BN from 'bn.js';
+
+// Class wrapping a plain object
+export abstract class Assignable {
+  constructor(properties: {[key: string]: any}) {
+    Object.keys(properties).forEach((key: string) => {
+      this[key] = properties[key];
+    });
+  }
+
+  encode(): Buffer {
+    return Buffer.from(borsh.serialize(SCHEMA, this));
+  }
+
+  static decode<T extends Assignable>(data: Buffer): T {
+    return borsh.deserializeUnchecked(SCHEMA, this, data);
+  }
+}
+
+// Class representing a Rust-compatible enum, since enums are only strings or
+// numbers in pure JS
+export abstract class Enum extends Assignable {
+  enum: string;
+  constructor(properties: any) {
+    super(properties);
+    if (Object.keys(properties).length !== 1) {
+      throw new Error('Enum can only take single value');
+    }
+    this.enum = '';
+    Object.keys(properties).forEach(key => {
+      this.enum = key;
+    });
+  }
+}
+
+export class Fee extends Assignable {
+  denominator: number;
+  numerator: number;
```

Comment:
These should be ok as number

Reply:
why is this OK as number?

Reply:
Although they're both `u64`s underneath, people *most likely* wouldn't use something so big.  To be totally safe, they should be `BN`s too, but it probably isn't important